source/dub/project.d

```diff
@@ -1243,17 +1223,30 @@ private string processVars(string var, in Project project, in Package pack, bool
 	} else return var;
 }
 
-private string getVariable(string name, in Project project, in Package pack)
+private string getVariable(Project, Package)(string name, in Project project, in Package pack)
 {
 	import std.process : environment;
-	if (name == "PACKAGE_DIR") return pack.path.toNativeString();
-	if (name == "ROOT_PACKAGE_DIR") return project.rootPackage.path.toNativeString();
+	Path path;
+	if (name == "PACKAGE_DIR")
+		path = pack.path;
+	else if (name == "ROOT_PACKAGE_DIR")
+		path = project.rootPackage.path;
 
 	if (name.endsWith("_PACKAGE_DIR")) {
 		auto pname = name[0 .. $-12];
 		foreach (prj; project.getTopologicalPackageList())
 			if (prj.name.toUpper().replace("-", "_") == pname)
```

Comment:
FWIW there's no need to allocate for this check,sth. like this should work fine and is @nogc:
`prj.name.asUpperCase.map!(a == '-' ? '_' : a).equal(pname)`